Orange/regression/pls.py

```diff
@@ -0,0 +1,190 @@
+from Orange.regression.base_regression import SklLearnerRegression, \
+    SklModelRegression
+
+__all__ = ["PLSRegressionLearner"]
+
+
+class _FeatureScorerMixin(LearnerScorer):
+    feature_type = Variable
+    class_type = ContinuousVariable
+
+    def score(self, data):
+        model = self(data)
+        return np.abs(model.coefficients), model.domain.attributes
+
+
+class _PLSCommonTransform:
+
+    def __init__(self, pls_model):
+        self.pls_model = pls_model
+
+    def _transform_with_numpy_output(self, X, Y):
+        """
+        # the next command does the following
+        x_center = X - pls._x_mean
+        y_center = Y - pls._y_mean
+        t = x_center @ pls.x_rotations_
+        u = y_center @ pls.y_rotations_
+        """
+        pls = self.pls_model.skl_model
+        t, u = pls.transform(X, Y)
+        return np.hstack((t, u))
+
+    def __call__(self, data):
+        if data.domain != self.pls_model.domain:
+            data = data.transform(self.pls_model.domain)
+        if len(data.Y.shape) == 1:
+            Y = data.Y.reshape(-1, 1)
+        else:
+            Y = data.Y
+        return self._transform_with_numpy_output(data.X, Y)
+
+    def __eq__(self, other):
+        if self is other:
+            return True
+        return type(self) is type(other) \
+            and self.pls_model == other.pls_model
+
+    def __hash__(self):
+        return hash(self.pls_model)
+
+
+class PLSProjector(SharedComputeValue):
+    InheritEq = True
+
+    def __init__(self, transform, feature):
+        super().__init__(transform)
+        self.feature = feature
+
+    def compute(self, _, shared_data):
+        return shared_data[:, self.feature]
+
+
+class PLSModel(SklModelRegression):
+    var_prefix_X = "PLS T"
+    var_prefix_Y = "PLS U"
+
+    @property
+    def coefficients(self):
+        return self.skl_model.coef_
+
+    def predict(self, X):
+        vals = self.skl_model.predict(X)
+        if len(self.domain.class_vars) == 1:
+            vals = vals.ravel()
+        return vals
+
+    def __str__(self):
+        return f"PLSModel {self.skl_model}"
+
+    def _get_var_names(self, n, prefix):
+        names = [f"{prefix}{postfix}" for postfix in range(1, n + 1)]
+        return get_unique_names([var.name for var in self.domain.metas], names)
+
+    def project(self, data):
+        if not isinstance(data, Table):
+            raise RuntimeError("PLSModel can only project tables")
+
+        transformer = _PLSCommonTransform(self)
+
+        def trvar(i, name):
+            return ContinuousVariable(name,
+                                      compute_value=PLSProjector(transformer,
+                                                                 i))
+
+        n_components = self.skl_model.x_loadings_.shape[1]
+
+        var_names_X = self._get_var_names(n_components, self.var_prefix_X)
+        var_names_Y = self._get_var_names(n_components, self.var_prefix_Y)
+
+        domain = Domain(
```

Comment:
Didn't we (with @lanzagar) speak about including the original data in this output too on Monday? Or did you change your mind?